src/lib/transform-fast-rest.js

```diff
@@ -88,3 +93,11 @@ export default function fastRestTransform({ template, types: t }) {
 		},
 	};
 }
+
+function getParentFunction(node, t) {
+	while ((node = node.parentPath)) {
+		if (t.isFunction(node)) {
```

Comment:
Arrow functions can access `arguments` from the enclosing scope, so we could allow them in the tree here:

```suggestion
		if (t.isFunction(node) && !t.isArrowFunctionExpression(node)) {
```

Reply:
I thought about that, but if the rest parameters transformation is applied with the same criteria of the arrow function transformation there would be no gain in terms of bundlesize.

Instead if the `[].slice.call` is moved in the root function scope it would be cached in some cases.

Reply:
I've made the change and without the arrow function check the bundle size of the test case seems smaller:

With the arrow function check:
```
157 B: parameters-rest-closure.js.gz
115 B: parameters-rest-closure.js.br
165 B: parameters-rest-closure.esm.js.gz
124 B: parameters-rest-closure.esm.js.br
247 B: parameters-rest-closure.umd.js.gz
189 B: parameters-rest-closure.umd.js.br
```

With getFunctionParent
```
146 B: parameters-rest-closure.js.gz
106 B: parameters-rest-closure.js.br
153 B: parameters-rest-closure.esm.js.gz
113 B: parameters-rest-closure.esm.js.br
238 B: parameters-rest-closure.umd.js.gz
177 B: parameters-rest-closure.umd.js.br
```

Reply:
makes sense from a Gzip perspective (repetition is essentially free), just there's a runtime cost associated with materializing `arguments` via slice() where it wouldn't otherwise be necessary.